### PR TITLE
Reject robots unless it is the app on pub.dartlang.org

### LINKED ISSUE #872: When the host is not pub.dartlang.org, have robots.txt block the whole domain
We're getting a crazy number of search hits against our non-primary domain.

We should just block them all.

If a request for `robots.txt` comes in and it's not for the `pub.dartlang.org` host, send a go-away response.

```
Disallow: /
```

### PR DESCRIPTION
**Pull request by isoos**
Fixes #872

### CODE REVIEW

File: app/lib/frontend/handlers.dart [hunk 1]
```diff
@@ -82,8 +82,13 @@ Future<shelf.Response> appHandler(
   } else if (path.startsWith(staticUrls.staticPath)) {
     return _staticsHandler(request);
   } else if (staticRootFiles.contains(path)) {
-    return _staticsHandler(request,
-        pathOverride: '${staticUrls.staticPath}$path');
+    print(path);
```
Comment (jakobr-google) [inline app/lib/frontend/handlers.dart]: Remove the print statements, or use debug logging?

File: app/lib/frontend/handlers.dart [hunk 2]
```diff
@@ -82,8 +82,13 @@ Future<shelf.Response> appHandler(
...
+    var pathOverride = '${staticUrls.staticPath}$path';
+    if (path == '/robots.txt' && !_isProd(request)) {
+      pathOverride = '${staticUrls.staticPath}/no_robots.txt';
```
Comment (jakobr-google) [inline app/lib/frontend/handlers.dart]: Could this be
```dart
return rejectRobotsHandler(request);
```
instead? (So we only have the reject robots.txt contents in one place.)

File: app/lib/dartdoc/handlers.dart [hunk 3]
```diff
@@ -17,6 +17,7 @@ Future<shelf.Response> dartdocServiceHandler(shelf.Request request) async {
   final path = request.requestedUri.path;
   final handler = {
     '/debug': debugHandler,
+    '/robots.txt': rejectRobotsHandler,
```
Comment (kevmoo) [inline app/lib/dartdoc/handlers.dart]: We might want to add a todo here about removing when this app has content...right?

**Comment by kevmoo** [comment]
@isoos – travis be sad

**Pull request merged**